pkg/gatewayserver/io/io.go

```diff
@@ -88,8 +88,17 @@ type Connection struct {
 	statusCh chan *ttnpb.GatewayStatus
 	txAckCh  chan *ttnpb.TxAcknowledgment
 
-	statsChangedCh chan struct{}
-	locCh          chan struct{}
+	upChangedCh     chan struct{}
+	downChangedCh   chan struct{}
+	statusChangedCh chan struct{}
+	locCh           chan struct{}
+}
+
+// Traffic denotes the kind of traffic.
+type Traffic struct {
```

Comment:
Now I wonder why this is necessary still, if you basically keep local state in the func

Reply:
Removed, updating based on timestamps as discussed.